### PR TITLE
fix(material): multi-select when passing observable data

### PR DESCRIPTION
**Pull request by juristr**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

fix when passing an observable input to the multi-select

**Please check if the PR fulfills these requirements**
- [x] The commit messages follow our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [x] A unit test has been written for this change.
- [x] Running `npm run build` produced a successful build. (Unit testing can be done by running `npm test`;)
- [x] My code has been linted. (`npm run lint` to do this. `npm run build` will fail if there are files not linted.)

### CODE REVIEW

File: src/material/select/src/select.spec.ts [hunk 1]
```diff
@@ -9,8 +9,9 @@ import { FormGroup, ReactiveFormsModule } from '@angular/forms';
 import { FormlyModule, FormlyForm } from '@ngx-formly/core';
 import { FormlySelectModule } from '@ngx-formly/core/select';
 import { FormlyFieldSelect } from './select.type';
-import { of as observableOf } from 'rxjs';
+import { of as observableOf, of } from 'rxjs';
```
Comment (aitboudad) [inline src/material/select/src/select.spec.ts]: `observableOf` = `of`

File: src/material/select/src/select.type.ts [hunk 2]
```diff
@@ -46,20 +50,20 @@ export class FormlyFieldSelect extends FieldType {
   };
 
   get value() { return this.formControl.value || []; }
-  get state() {
+  getState(options: any[]) {
     if (this.value.length > 0) {
-      return this.value.length !== this.formFieldControl.options.length
```
Comment (aitboudad) [inline src/material/select/src/select.type.ts]: `formFieldControl` declaration is not used anymore and can be removed too:
```patch
-- @ViewChild(MatSelect) formFieldControl!: MatSelect;
```

File: src/material/select/src/select.spec.ts (hunk 1 above)
Comment (juristr) [reply src/material/select/src/select.spec.ts]: oh yes, probably auto-imported by VSCode 👍

**Comment by aitboudad** [comment]
Thank you @juristr!

**Pull request merged**